pandas/core/indexes/period.py

```diff
@@ -387,17 +475,28 @@ def asof_locs(self, where, mask):
 
         return result
 
+    def _box_values_as_index(self):
+        """
+        return object Index which contains boxed values
+        """
+        # TODO(DatetimeArray): remove
+        # Have to add our name.
+        return Index(self._data._box_values_as_index(), name=self.name)
```

Comment:
Recently implemented wrap_array_method.  I think the thing to do here is edit that function so that this becomes a one-liner `_box_values_as_index = wrap_array_method("_box_values_as_index", pin_name=True)`

Reply:
BTW, how can this actually work, as PeriodArray has no such method?